### DISCUSSION TITLE
Detect idle status to shutdown openvscode-server when there is no active connection

### DISCUSSION THREAD

**Discussion by pyhp2017** [gitpod-io/openvscode-server#431] [General]
I need a way to detect that if there is any active connection in the IDE or not (to kill it!). it could be possible by adding an endpoint to server, for example ```localhost:3000/heartbeat``` and it should return a json and says what was the last connection time and is there any active connection or not.

Thanks!

**Comment by jeanp413** [comment]
@pyhp2017 this is already built-in in the server, you can use `--enable-remote-auto-shutdown` (will wait 5 min after last connection closes) and `--remote-auto-shutdown-without-delay` when launching the server
